import os
from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(1280, 730)
        MainWindow.setLayoutDirection(QtCore.Qt.LeftToRight)
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.verticalLayout_4 = QtWidgets.QVBoxLayout(self.centralwidget)
        self.verticalLayout_4.setObjectName("verticalLayout_4")
        self.horizontalLayout_3 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_3.setSizeConstraint(QtWidgets.QLayout.SetMinimumSize)
        self.horizontalLayout_3.setObjectName("horizontalLayout_3")
        self.groupBox = QtWidgets.QGroupBox(self.centralwidget)
        font = QtGui.QFont()
        font.setPointSize(10)
        self.groupBox.setFont(font)
        self.groupBox.setAlignment(QtCore.Qt.AlignCenter)
        self.groupBox.setObjectName("groupBox")
        self.horizontalLayout_2 = QtWidgets.QHBoxLayout(self.groupBox)
        self.horizontalLayout_2.setContentsMargins(-1, 8, -1, 9)
        self.horizontalLayout_2.setObjectName("horizontalLayout_2")
        self.verticalLayout_2 = QtWidgets.QVBoxLayout()
        self.verticalLayout_2.setObjectName("verticalLayout_2")
        self.label_InputView = QtWidgets.QLabel(self.groupBox)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(1)
        sizePolicy.setVerticalStretch(1)
        sizePolicy.setHeightForWidth(self.label_InputView.sizePolicy().hasHeightForWidth())
        self.label_InputView.setSizePolicy(sizePolicy)
        self.label_InputView.setMinimumSize(QtCore.QSize(280, 280))
        self.label_InputView.setText("")
        # self.label_InputView.setScaledContents(True)
        self.label_InputView.setAlignment(QtCore.Qt.AlignCenter)
        self.label_InputView.setObjectName("label_InputView")
        self.verticalLayout_2.addWidget(self.label_InputView)
        self.verticalLayout = QtWidgets.QVBoxLayout()
        self.verticalLayout.setObjectName("verticalLayout")
        self.line = QtWidgets.QFrame(self.groupBox)
        self.line.setFrameShape(QtWidgets.QFrame.HLine)
        self.line.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line.setObjectName("line")
        self.verticalLayout.addWidget(self.line)
        self.horizontalLayout = QtWidgets.QHBoxLayout()
        self.horizontalLayout.setObjectName("horizontalLayout")
        self.label_Mode = QtWidgets.QLabel(self.groupBox)
        self.label_Mode.setAlignment(QtCore.Qt.AlignCenter)
        self.label_Mode.setObjectName("label_Mode")
        self.horizontalLayout.addWidget(self.label_Mode)
        self.line_5 = QtWidgets.QFrame(self.groupBox)
        self.line_5.setFrameShape(QtWidgets.QFrame.VLine)
        self.line_5.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_5.setObjectName("line_5")
        self.horizontalLayout.addWidget(self.line_5)
        self.comboBox_ModeSelect = QtWidgets.QComboBox(self.groupBox)
        self.comboBox_ModeSelect.setObjectName("comboBox_ModeSelect")
        self.comboBox_ModeSelect.addItem("")
        self.comboBox_ModeSelect.addItem("")
        self.horizontalLayout.addWidget(self.comboBox_ModeSelect)
        self.verticalLayout.addLayout(self.horizontalLayout)
        self.line_2 = QtWidgets.QFrame(self.groupBox)
        self.line_2.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_2.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_2.setObjectName("line_2")
        self.verticalLayout.addWidget(self.line_2)
        self.pushButton_LoadFile = QtWidgets.QPushButton(self.groupBox)
        self.pushButton_LoadFile.setObjectName("pushButton_LoadFile")
        self.verticalLayout.addWidget(self.pushButton_LoadFile)
        self.line_3 = QtWidgets.QFrame(self.groupBox)
        self.line_3.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_3.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_3.setObjectName("line_3")
        self.verticalLayout.addWidget(self.line_3)
        self.pushButton_Run_Stop = QtWidgets.QPushButton(self.groupBox)
        self.pushButton_Run_Stop.setObjectName("pushButton_Run_Stop")
        self.verticalLayout.addWidget(self.pushButton_Run_Stop)
        self.line_4 = QtWidgets.QFrame(self.groupBox)
        self.line_4.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_4.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_4.setObjectName("line_4")
        self.verticalLayout.addWidget(self.line_4)
        self.verticalLayout_2.addLayout(self.verticalLayout)
        self.textBrowser_Log = QtWidgets.QTextBrowser(self.groupBox)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.textBrowser_Log.sizePolicy().hasHeightForWidth())
        self.textBrowser_Log.setSizePolicy(sizePolicy)
        self.textBrowser_Log.setMinimumSize(QtCore.QSize(280, 0))
        self.textBrowser_Log.setObjectName("textBrowser_Logs")
        self.verticalLayout_2.addWidget(self.textBrowser_Log)
        self.verticalLayout_2.setStretch(0, 3)
        self.verticalLayout_2.setStretch(1, 2)
        self.verticalLayout_2.setStretch(2, 2)
        self.horizontalLayout_2.addLayout(self.verticalLayout_2)
        self.horizontalLayout_3.addWidget(self.groupBox)
        self.groupBox_2 = QtWidgets.QGroupBox(self.centralwidget)
        font = QtGui.QFont()
        font.setPointSize(10)
        self.groupBox_2.setFont(font)
        self.groupBox_2.setLayoutDirection(QtCore.Qt.LeftToRight)
        self.groupBox_2.setAlignment(QtCore.Qt.AlignCenter)
        self.groupBox_2.setObjectName("groupBox_2")
        self.gridLayout = QtWidgets.QGridLayout(self.groupBox_2)
        self.gridLayout.setContentsMargins(12, 8, 12, 9)
        self.gridLayout.setObjectName("gridLayout")
        self.label_DetectView = QtWidgets.QLabel(self.groupBox_2)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(1)
        sizePolicy.setVerticalStretch(1)
        sizePolicy.setHeightForWidth(self.label_DetectView.sizePolicy().hasHeightForWidth())
        self.label_DetectView.setSizePolicy(sizePolicy)
        self.label_DetectView.setMinimumSize(QtCore.QSize(640, 640))
        self.label_DetectView.setText("")
        # self.label_DetectView.setScaledContents(True)
        self.label_DetectView.setAlignment(QtCore.Qt.AlignCenter)
        self.label_DetectView.setObjectName("label_DetectView")
        self.gridLayout.addWidget(self.label_DetectView, 0, 0, 1, 1)
        self.horizontalLayout_3.addWidget(self.groupBox_2)
        self.groupBox_3 = QtWidgets.QGroupBox(self.centralwidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Preferred, QtWidgets.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(1)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.groupBox_3.sizePolicy().hasHeightForWidth())
        self.groupBox_3.setSizePolicy(sizePolicy)
        font = QtGui.QFont()
        font.setPointSize(10)
        self.groupBox_3.setFont(font)
        self.groupBox_3.setAlignment(QtCore.Qt.AlignCenter)
        self.groupBox_3.setObjectName("groupBox_3")
        self.verticalLayout_3 = QtWidgets.QVBoxLayout(self.groupBox_3)
        self.verticalLayout_3.setContentsMargins(-1, 8, -1, 9)
        self.verticalLayout_3.setObjectName("verticalLayout_3")
        self.textBrowser_Result = QtWidgets.QTextBrowser(self.groupBox_3)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
        sizePolicy.setHorizontalStretch(1)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.textBrowser_Result.sizePolicy().hasHeightForWidth())
        self.textBrowser_Result.setSizePolicy(sizePolicy)
        self.textBrowser_Result.setMinimumSize(QtCore.QSize(240, 0))
        self.textBrowser_Result.setObjectName("textBrowser_Result")
        self.verticalLayout_3.addWidget(self.textBrowser_Result)
        self.horizontalLayout_3.addWidget(self.groupBox_3)
        self.horizontalLayout_3.setStretch(0, 1)
        self.horizontalLayout_3.setStretch(1, 2)
        self.horizontalLayout_3.setStretch(2, 1)
        self.verticalLayout_4.addLayout(self.horizontalLayout_3)
        MainWindow.setCentralWidget(self.centralwidget)
        self.statusbar = QtWidgets.QStatusBar(MainWindow)
        self.statusbar.setObjectName("statusbar")
        MainWindow.setStatusBar(self.statusbar)
        # 在setupUi的最后添加样式表加载代码
        self.load_stylesheet(MainWindow)
        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def load_stylesheet(self, window):
        """加载外部样式表"""
        try:
            style_path = os.path.join(os.path.dirname(__file__), 's_1.qss')
            # 读取样式表文件
            with open(style_path, 'r', encoding='utf-8') as f:
                style = f.read()
                window.setStyleSheet(style)
        except Exception as e:
            print(f"无法加载样式表: {e}")
            # 使用默认样式
            window.setStyleSheet("")

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "基于深度学习的手势识别系统"))
        self.groupBox.setTitle(_translate("groupBox", "操作"))
        self.label_Mode.setText(_translate("label_Mode", "选择模式"))
        self.comboBox_ModeSelect.setItemText(0, _translate("comboBox_ModeSelect", "图片检测"))
        self.comboBox_ModeSelect.setItemText(1, _translate("comboBox_ModeSelect", "视频检测"))
        self.pushButton_LoadFile.setText(_translate("pushButton_LoadFile", "加载文件"))
        self.pushButton_Run_Stop.setText(_translate("pushButton_Run_Stop", "开始检测"))
        self.groupBox_2.setTitle(_translate("groupBox_2", "预测"))
        self.groupBox_3.setTitle(_translate("groupBox_3", "记录"))
